### PR TITLE
API provides notification to slack on incoming mail

### PR DESCRIPTION
**Pull request by oliverochman**
PT https://www.pivotaltracker.com/story/show/178449305

### User Story
```
As an API
In order to notify Flex Coast about new inquiries
I want to send new inquiries off to slack
```
### Changes proposed 
- Send notification to brokers slack channel when visitor submits new inquiry

### Packages/Gems added
- gem 'webmock'
- gem 'slack-notify'

### What I have learned working on this feature
- How to use slack-notify gem.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,13 +9,16 @@ gem 'puma', '~> 5.0'
 gem 'bootsnap', '>= 1.4.4', require: false
 gem 'rack-cors', require: 'rack/cors'
 gem 'active_model_serializers'
+gem "slack-notify"
```
Comment (DavDan1) [inline Gemfile]: solargraph :)

File: spec/requests/api/visitors_can_create_inquiries_spec.rb [hunk 2]
```diff
@@ -66,7 +71,7 @@
         end
 
         it 'phone' do
-          expect(mail_delivery[0].body).to include(0707123456)
+          expect(mail_delivery[0].body).to include(0o707123456)
```
Comment (lukeperrin10) [inline spec/requests/api/visitors_can_create_inquiries_spec.rb]: Just going to point this one out.... know it came up in the last one too.

Comment (1CIM) [reply spec/requests/api/visitors_can_create_inquiries_spec.rb]: it is the bloody blod solargraph...

Comment (tochman) [reply spec/requests/api/visitors_can_create_inquiries_spec.rb]: Again, change it to a string.... Pick the low-hanging fruit and move on...

**Review by DavDan1 — APPROVED** [review]

**Review by arunbhalli — APPROVED** [review]

File: spec/requests/api/visitors_can_create_inquiries_spec.rb [hunk 3]
```diff
@@ -71,7 +71,7 @@
         end
 
         it 'phone' do
-          expect(mail_delivery[0].body).to include(0o707123456)
+          expect(mail_delivery[0].body).to include(0707123456)
```
Comment (tochman) [inline spec/requests/api/visitors_can_create_inquiries_spec.rb]: Please make this a string.

**Review by tochman — CHANGES_REQUESTED** [review]
The phone number needs to be a string

File: app/services/notification_service.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+require 'slack-notify'
+
+module NotificationService
+  def self.new_inquiry(inquiry)
+    InquiryMailer.created_email(inquiry).deliver
+    client = SlackNotify::Client.new(
+      webhook_url: Rails.application.credentials.dig(:slack, :webhook_url)
+    )
```
Comment (LucasKnudsen) [inline app/services/notification_service.rb:8]: How can the webhook_url use itself as an argument in the `dig()` ? :o

**Review by LucasKnudsen — APPROVED** [review]

**Pull request merged**